### PR TITLE
Replace Issue Templates with Issue Forms

### PR DESCRIPTION
**Pull request by tnorling**
Replaces markdown issue templates with the new issue forms!

Also updates the bot to support this change.

Checkout a live instance of these forms on [my fork](https://github.com/tnorling/microsoft-authentication-library-for-js/issues/new/choose)

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:37]: Incrementally, we should update the placeholder to indicate the latest version of each library.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:52]: We should consider making these required and adding a "not applicable" option.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:29]:
```suggestion
      - "@azure/msal-node"
      - "passport-azure-ad"
      - "adal.js"
```

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:46]:
```suggestion
      - "@azure/msal-react"
      - "adal-angular"
```

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [reply .github/ISSUE_TEMPLATE/bug_report.yml:29]: I guess these can be added later once their PRs have been merged

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:115]:
```suggestion
      - Azure AD / MSA
```

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (jasonnutter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:102]: I think we should users to indicate if they have tried reproducing this issues with the latest version of their library.

**Review by jo-arroyo — APPROVED** [review]
Approved, but wondering if we should leave an (optional) area for them to include logs?

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (sameerag) [inline .github/ISSUE_TEMPLATE/bug_report.yml:125]: There should be N/A section - or may be make it `Server` use case, `msal-node`, `adal-node`, `passport-js` do not have active browser in all use cases.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (sameerag) [inline .github/ISSUE_TEMPLATE/bug_report.yml:146]: Should we have optional customer product/company? May work with stats later. Just an idea.

File: .github/ISSUE_TEMPLATE/documentation.yml
Comment (sameerag) [inline .github/ISSUE_TEMPLATE/documentation.yml:36]: We can also give them an option to open a PR on our documentation and link it to this issue template, encouraging community contributions.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml:59]: Done

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml:29]: No harm in getting ahead of it. Done.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml:49]: Done

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Done

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml:139]: Good callout, done!

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml:161]: Will take this incrementally if we see value in it later. Right now we only care so we know whether or not we can follow up over Teams

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/bug_report.yml:116]: What if we update the prompt for the version field to say "Please indicate the latest version this can be reproduced in" and add a feature to the bot incrementally that auto comments on issues with a link to npm/changelog if they indicate they are using an older version? I'm not sure I see a whole lot of value in a simple checkbox tbh

File: .github/ISSUE_TEMPLATE/documentation.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/documentation.yml:43]: > Approved, but wondering if we should leave an (optional) area for them to include logs? Otherwise, very exciting!

Great call, added! There's also a generic comment box at the end that people can fill in with misc. information that may not be covered in the form.

**Comment by tnorling** [comment]
> Approved, but wondering if we should leave an (optional) area for them to include logs? Otherwise, very exciting!

Great call, added! There's also a generic comment box at the end that people can fill in with misc. information that may not be covered in the form.

File: .github/ISSUE_TEMPLATE/documentation.yml
Comment (tnorling) [reply .github/ISSUE_TEMPLATE/documentation.yml:56]: Great idea! Added a markdown section to the top of this template suggesting PRs

File: .github/ISSUE_TEMPLATE/feature_request.yml
Comment (sameerag) [inline .github/ISSUE_TEMPLATE/feature_request.yml:46]: Not linked to this but related - we need to re-haul our roadmap structure for minor feature additions too, since some of them do not reflect in roadmap and we have multiple PRs open.

**Review by pkanher617 — APPROVED** [review]

**Review by sameerag — APPROVED** [review]
Lgtm. Thanks for addressing all comments. The open ones can be incrementally added.

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**
